app/wasm/test/queries_test.go

```diff
@@ -0,0 +1,489 @@
+package wasm
+
+import (
+	"fmt"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	wasmbindings "github.com/osmosis-labs/osmosis/v7/app/wasm/bindings"
+	"github.com/stretchr/testify/assert"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/osmosis-labs/osmosis/v7/app/wasm"
+)
+
+func TestFullDenom(t *testing.T) {
```

Comment:
This file includes many table tests on the go functions (not touching the contract) to cover many edge cases faster. We ensure the basic happy/sad paths are covered with the contract to ensure the data is parsed properly, but this allows quicker table testing.